gobblin-api/src/main/java/org/apache/gobblin/source/workunit/WorkUnit.java

```diff
@@ -82,6 +82,7 @@ public WorkUnit(SourceState state, Extract extract) {
 
     if (extract != null) {
       this.extract = extract;
+      super.addAll(extract);
```

Comment:
Why this change instead of leaving the code that propagates only the table name to the workunit? This may bloat up the state which is already large for gobblin-kafka. Also you would have to add this the constructors that don't call this one. Unless there is a reason for this, I would prefer to leave it the way it was to avoid unexpected side effects.